File: frontend/website/src/models/observation.js
// Model of Observation that match with the API's Model

class Observation {
    constructor(
      id,
      sensor_name,
      observable_property_name,
      time_start,
      time_end,
      value_int,
      value_float,
      value_bool,
      value_str
    ) {
      this.id = id
      this.sensor_name = sensor_name
      this.observable_property_name = observable_property_name
      this.time_start = time_start
      this.time_end = time_end
      this.value_int = value_int
      this.value_float = value_float
      this.value_bool = value_bool
      this.value_str = value_str
    }
  }
  
  export default Observation

File: backend/app/schemas/observation.py
import datetime
from typing import Optional, List, TYPE_CHECKING

from sqlmodel import Column, DateTime, Field, SQLModel, Relationship

from app.schemas.actuation_observation_link import ActuationObservationLink

if TYPE_CHECKING:
    from app.schemas.actuation import Actuation


class Observation(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    time_start: datetime.datetime = Field(sa_column=Column(DateTime(timezone=True),
                                                           nullable=False))

    time_end: Optional[datetime.datetime] = Field(sa_column=Column(DateTime(timezone=True)))

    # TODO Improve handling of the value system
    value_int: Optional[int]
    value_float: Optional[float]
    value_bool: Optional[bool]
    value_str: Optional[str]

    # Relations
    observable_property_name: str = Field(nullable=False, foreign_key="observableproperty.name")
    sensor_name: str = Field(nullable=False, foreign_key="sensor.name")
    actuations: List["Actuation"] = Relationship(back_populates="observations",
                                                 link_model=ActuationObservationLink)
    #
